recirq/quantum_chess/quantum_board.py

```diff
@@ -110,14 +117,12 @@ def with_state(self, basis_state: int) -> "CirqBoard":
         self.circuit = cirq.Circuit()
         self.ancilla_count = 0
         self.move_history = []
-        self.full_squares = basis_state
-        self.empty_squares = 0
-        for i in range(64):
-            self.empty_squares = set_nth_bit(
-                i, self.empty_squares, not nth_bit_of(i, self.full_squares)
-            )
-        # Each entry is a 2-tuple of (repetitions, probabilities) corresponding to the probabilities after each move.
-        self.move_history_probabilities_cache = []
+        initial_square_probs = (float(nth_bit_of(i, basis_state)) for i in range(64))
+        # Invariant:
+        # self.move_history_probabilities_cache == len(self.move_history) + 1
+        self.move_history_probabilities_cache = [
+            self._make_probability_history(10 ** 9, initial_square_probs)
```

Comment:
10**9 should also be a constant I think.  I doubt people will request more than 1 billion reps of the initial state, but it would still be good to list that as a constant at the top.